addons/web/static/src/model/record.js

```diff
@@ -167,17 +167,21 @@ export class Record extends Component {
         "onWillSaveRecord?",
     ];
     setup() {
-        if (this.props.fields) {
-            this.fields = this.props.fields;
+        const { activeFields, fieldNames, fields, resModel } = this.props;
+        if (!activeFields && !fieldNames) {
+            throw Error(`Record props should have either a "resModel" key or a "fieldNames" key`);
+        }
+        if (!fields && (!fieldNames || !resModel)) {
+            throw Error(
+                `Record props should have either a "fields" key or a "fieldNames" and a "resModel" key`
+            );
+        }
```

Comment:
shouldn't this be done by the props validation mechanism? If we can't access other props in the validate function, it's maybe something missing in owl

Reply:
I think you should create an owl issue